telegram/_bot.py

```diff
@@ -5838,12 +6076,104 @@ async def delete_sticker_from_set(
+
+        Raises:
+            :class:`telegram.error.TelegramError`
+
+        """
+        data: JSONDict = {"name": name}
+        result = await self._post(
+            "deleteStickerSet",
+            data,
+            read_timeout=read_timeout,
+            write_timeout=write_timeout,
+            connect_timeout=connect_timeout,
+            pool_timeout=pool_timeout,
+            api_kwargs=api_kwargs,
+        )
+        return result
+
+    @_log
+    async def set_sticker_set_thumbnail(
+        self,
+        name: str,
+        user_id: Union[str, int],
+        thumbnail: FileInput = None,
+        # Deprecated param `thumb` not included. Would be strange to allow someone to call
+        # the new method with an old param.
+        *,
+        read_timeout: ODVInput[float] = DEFAULT_NONE,
+        write_timeout: ODVInput[float] = DEFAULT_NONE,
+        connect_timeout: ODVInput[float] = DEFAULT_NONE,
+        pool_timeout: ODVInput[float] = DEFAULT_NONE,
+        api_kwargs: JSONDict = None,
+    ) -> bool:
+        """Use this method to set the thumbnail of a sticker set. Animated thumbnails can be set
+        for animated sticker sets only. Video thumbnails can be set only for video sticker sets
+        only.
+
+        .. versionadded:: NEXT.VERSION
+
+        Args:
+            name (:obj:`str`): Sticker set name
+            user_id (:obj:`int`): User identifier of created sticker set owner.
+            thumbnail (:obj:`str` | :term:`file object` | :obj:`bytes` | :class:`pathlib.Path`, \
+                optional): A **PNG** image with the thumbnail, must
+                be up to 128 kilobytes in size and have width and height exactly 100px, or a
+                **TGS** animation with the thumbnail up to 32 kilobytes in size; see
+                https://core.telegram.org/stickers#animation-requirements for animated
+                sticker technical requirements, or a **WEBM** video with the thumbnail up to 32
+                kilobytes in size; see
+                https://core.telegram.org/stickers#video-requirements for video sticker
+                technical requirements.
+                |fileinput|
+                Animated sticker set thumbnails can't be uploaded via HTTP URL.
+        Returns:
+            :obj:`bool`: On success, :obj:`True` is returned.
+
+        Raises:
+            :class:`telegram.error.TelegramError`
+
+        """
+        return await self._set_sticker_set_thumbnail(
+            name=name,
+            user_id=user_id,
+            thumbnail=thumbnail,
+            read_timeout=read_timeout,
+            write_timeout=write_timeout,
+            connect_timeout=connect_timeout,
+            pool_timeout=pool_timeout,
+            api_kwargs=api_kwargs,
+        )
+
     @_log
     async def set_sticker_set_thumb(
         self,
         name: str,
         user_id: Union[str, int],
         thumb: FileInput = None,
+        thumbnail: FileInput = None,
```

Comment:
I see no reason to add this argument. Changing from `thumb` to `thumbnail` won't save the user from changing to `set_sticker_set_thumbnail` … Issuing a deprecation warning for the method itself ofc is very sane.

Reply:
OK yes, that makes perfect sense. Done.